module/VuFind/src/VuFind/Controller/ContentController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Content Controller
+ *
+ * PHP Version 5
+ *
+ * Copyright (C) Villanova University 2011.
+ * Copyright (C) The National Library of Finland 2014-2016.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA    02111-1307    USA
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+namespace VuFind\Controller;
+
+/**
+ * Controller for mostly static pages that doesn't fall under any particular
+ * function.
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+class ContentController extends \VuFind\Controller\AbstractBase
+{
+    /**
+     * Default action if none provided
+     *
+     * @return Zend\View\Model\ViewModel
+     */
+    public function contentAction()
+    {
+        $event      = $this->getEvent();
+        $routeMatch = $event->getRouteMatch();
+        $page       = strtolower($routeMatch->getParam('page'));
+        $themeInfo  = $this->getServiceLocator()->get('VuFindTheme\ThemeInfo');
+        $translator = $this->getServiceLocator()->get('VuFind\Translator');
+        $language   = $translator->getLocale();
+        $defaultLanguage = $this->getConfig()->Site->language;
+
+        // Try to find a template using
+        // 1.) Current language suffix
+        // 2.) Default language suffix
+        // 3.) No language suffix
+        if (null !== $themeInfo->findContainingTheme(
+            "templates/content/{$page}_$language.phtml"
+        )) {
+            $page = "{$page}_$language";
+        } elseif (null !== $themeInfo->findContainingTheme(
+            "templates/content/{$page}_$defaultLanguage.phtml"
+        )) {
+            $page = "{$page}_$defaultLanguage";
+        }
```

Comment:
There's a legitimate use case for non-language specific templates. If you have short pages with content only in, say, two languages, you might want to have both languages in the same template and just check for `$this->layout()->userLang`. Or you could have the textual content in translation strings as the page might mainly consist of e.g. images. 

I believe that falling back to 'en' is not the right thing to do in this case. Our institutions usually have English enabled, but then it's 'en-gb', not 'en', and the primary language is 'fi' or 'sv'.